DashAI/back/api/api_v1/endpoints/explainers.py

```diff
@@ -0,0 +1,619 @@
+async def get_global_explainers(
+    run_id: int,
+    session_factory: Callable[..., ContextManager[Session]] = Depends(
+        Provide[Container.db.provided.session]
+    ),
+):
+    """Returns the available global explanainers in the database associated with the
+    run_id.
+
+    Parameters
+    ----------
+    run_id: int
+        Run id to select the global explanations to retrieve.
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A list of dicts containing global explainers.
+
+    Raises
+    ------
+    HTTPException
+        If there are no global explainers associated with the run_id in the DB.
+    """
+    with session_factory() as db:
+        try:
+            global_explainers = db.scalars(
+                select(GlobalExplainer).where(GlobalExplainer.run_id == run_id)
+            ).all()
+
+        except exc.SQLAlchemyError as e:
+            log.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return global_explainers
+
+
+@router.get("/global/explanation/{explainer_id}")
+@inject
+async def get_global_explanation(
+    explainer_id: int,
+    component_registry: ComponentRegistry = Depends(
+        Provide[Container.component_registry]
+    ),
+    session_factory: Callable[..., ContextManager[Session]] = Depends(
+        Provide[Container.db.provided.session]
+    ),
+):
+    """Returns the global explanation associated with id explainer_id.
+
+    Parameters
+    ----------
+    explaniner_id: int
+        Id to select the global explanation to retrieve.
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A JSON with the explanation.
+
+    Raises
+    ------
+    HTTPException
+        If there is no global explanation associated with the explanation_id in the
+        database.
+    """
+    with session_factory() as db:
+        try:
+            global_explainer = db.scalars(
+                select(GlobalExplainer).where(GlobalExplainer.id == explainer_id)
+            ).all()
+
+            if not global_explainer:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explainer not found",
+                )
+
+            if global_explainer[0].status != ExplainerStatus.FINISHED:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Explaination not found",
+                )
```

Comment:
I think this is a good way to manage errors, but I have a doubt, what if I want to see for example an explainer that has failed?

Reply:
note that maybe it is not necessary to fix this in this pr.